soc/xtensa/esp32/soc.c

```diff
@@ -84,3 +84,51 @@ int arch_printk_char_out(int c)
 	esp32_rom_uart_tx_one_char(c);
 	return 0;
 }
+
+
+const struct esp32_peripheral esp32_peripherals[] = {
+	/* DPORT_PERIP */
+	{DPORT_PERIP_CLK_EN_REG, DPORT_PERIP_RST_EN_REG, DPORT_TIMERS_CLK_EN },
+	{DPORT_PERIP_CLK_EN_REG, DPORT_PERIP_RST_EN_REG, DPORT_SPI01_CLK_EN },
+	{DPORT_PERIP_CLK_EN_REG, DPORT_PERIP_RST_EN_REG, DPORT_UART_CLK_EN },
```

Comment:
I don't understand what's happening here or why this is related.  Is this an Espressif device enumeration or something?  if so, it probably wants to be integrated with devicetree instead.

Reply:
This is mainly used to control Reset and Clock Enable of each peripheral inside the chip.
This table is used by `clock_control_on` and `clock_control_off` to turn a peripheral on/off.
The enum values in `soc.h` [line: 65](https://github.com/zephyrproject-rtos/zephyr/pull/25394/files#diff-737ec6c8bb02fda721d25e320f1418c2R65) are used in conjunction with this table.

Example use case:
- Call `clock_control_on(dev, PERIPH_UART0_MODULE)` 
- The `clock_control_on` function will use `esp32_peripherals[PERIPH_UART0_MODULE]` to get Reset Register, Clock Register and the Bit offset for this peripheral.
- Clear 'reset' bit and set `clock` bit.

I will check if there is a DeviceTree example for this case.

I'm open for any suggestion to minimize this table, or implement a better solution.

Reply:
Seems to me like you want to put that register offset inside the device config struct then, so you pay the cost for the devices you actually use in your image and don't have to store a giant table of all of them?

Reply:
I have updated the PR now.
I replaced that large table with Module IDs in `esp32_clock.h`.
Those IDs are actually offsets in CLK/RST Registers.

And since clk/rst registers are not a contagious block with consecutive addresses
I made an Array with their addresses `clock_control_regs`.

So the current implementation use the following formulas to determine which register/offset should use to turn a peripheral on/off:
`RegisterBank Index = ModuleID / 32`
`offset =  ModuleID % 32`